tests/js/spec/components/hovercard.spec.tsx

```diff
@@ -0,0 +1,155 @@
+    expect(document.querySelectorAll(`#${HOVERCARD_PORTAL_ID}`)).toHaveLength(1);
+  });
+  it('Displays card', async () => {
+    mountWithTheme(
+      <Hovercard
+        position="top"
+        body="Hovercard Body"
+        header="Hovercard Header"
+        displayTimeout={0}
+      >
+        Hovercard Trigger
+      </Hovercard>
+    );
+
+    userEvent.hover(screen.getByText('Hovercard Trigger'));
+
+    expect(await screen.findByText(/Hovercard Body/)).toBeInTheDocument();
+    expect(await screen.findByText(/Hovercard Header/)).toBeInTheDocument();
+  });
+
+  it('Does not display card', () => {
+    mountWithTheme(
+      <Hovercard
+        position="top"
+        body="Hovercard Body"
+        header="Hovercard Header"
+        displayTimeout={0}
+        show={false}
+      >
+        Hovercard Trigger
+      </Hovercard>
+    );
+
+    userEvent.hover(screen.getByText('Hovercard Trigger'));
+    jest.runAllTimers();
+
+    expect(screen.queryByText(/Hovercard Body/)).not.toBeInTheDocument();
+    expect(screen.queryByText(/Hovercard Header/)).not.toBeInTheDocument();
+  });
+
+  it('Always displays card', () => {
+    mountWithTheme(
+      <Hovercard
+        position="top"
+        body="Hovercard Body"
+        header="Hovercard Header"
+        displayTimeout={0}
+        show
+      >
+        Hovercard Trigger
+      </Hovercard>
+    );
+
+    expect(screen.getByText(/Hovercard Body/)).toBeInTheDocument();
+    expect(screen.getByText(/Hovercard Header/)).toBeInTheDocument();
+  });
+
+  it('Respects displayTimeout displays card', () => {
+    mountWithTheme(
+      <Hovercard
+        position="top"
+        body="Hovercard Body"
+        header="Hovercard Header"
+        displayTimeout={100}
+      >
+        Hovercard Trigger
+      </Hovercard>
+    );
+
+    userEvent.hover(screen.getByText('Hovercard Trigger'));
+
+    act(() => {
+      jest.advanceTimersByTime(99);
+    });
+
+    expect(screen.queryByText(/Hovercard Body/)).not.toBeInTheDocument();
+    expect(screen.queryByText(/Hovercard Header/)).not.toBeInTheDocument();
+
+    act(() => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.getByText(/Hovercard Body/)).toBeInTheDocument();
+    expect(screen.getByText(/Hovercard Header/)).toBeInTheDocument();
+  });
+
+  it('Doesnt leak timeout', () => {
+    mountWithTheme(
+      <Hovercard
+        position="top"
+        body="Hovercard Body"
+        header="Hovercard Header"
+        displayTimeout={100}
+      >
+        Hovercard Trigger
+      </Hovercard>
+    );
+
+    userEvent.hover(screen.getByText('Hovercard Trigger'));
+
+    act(() => {
+      jest.advanceTimersByTime(99);
+    });
+
+    expect(screen.queryByText(/Hovercard Body/)).not.toBeInTheDocument();
+    expect(screen.queryByText(/Hovercard Header/)).not.toBeInTheDocument();
+
+    userEvent.unhover(screen.getByText('Hovercard Trigger'));
+
+    act(() => {
+      jest.advanceTimersByTime(1);
+    });
```

Comment:
do you think we can replace the `jest.advanceTimersByTime` with ` await screen.*`?

Reply:
I think we should keep this one as is because we are testing that hovercard shows up exactly on time :)